framework/DataObjects/XDataSet.py

```diff
@@ -1124,7 +1124,7 @@ def _fromCSV(self,fileName,**kwargs):
         data = df[[var,self.sampleTag]].groupby(self.sampleTag).first().values[:,0]
         dtype = self._getCompatibleType(data.item(0))
         arrays[var] = self._collapseNDtoDataArray(data,var,labels=samples,dtype=dtype)
-    self._convertArrayListToDataset(arrays,action='replace')
+    self._convertArrayListToDataset(arrays,action='extend')
```

Comment:
where do you check if the new data are consistent with the old ones? It might cause an unexpected behavior

Reply:
Oh...I see Congjian and you already found the issue

Reply:
The error isn't too bad, actually, it says that the expected variables weren't found found in <filename>.